src/libraries/Microsoft.Extensions.Logging.Console/src/AnsiParser.cs

```diff
@@ -0,0 +1,207 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class AnsiParser
+    {
+        private readonly Action<string, int, int, ConsoleColor?, ConsoleColor?> _onParseWrite;
+        public AnsiParser(Action<string, int, int, ConsoleColor?, ConsoleColor?> onParseWrite)
+        {
+            if (onParseWrite == null)
+            {
+                throw new ArgumentNullException(nameof(onParseWrite));
+            }
+            _onParseWrite = onParseWrite;
+        }
+
+        /// <summary>
+        /// Parses a subset of display attributes
+        /// Set Display Attributes
+        /// Set Attribute Mode [{attr1};...;{attrn}m
+        /// Sets multiple display attribute settings. The following lists standard attributes that are getting parsed:
+        /// 1 Bright
+        /// Foreground Colours
+        /// 30 Black
+        /// 31 Red
+        /// 32 Green
+        /// 33 Yellow
+        /// 34 Blue
+        /// 35 Magenta
+        /// 36 Cyan
+        /// 37 White
+        /// Background Colours
+        /// 40 Black
+        /// 41 Red
+        /// 42 Green
+        /// 43 Yellow
+        /// 44 Blue
+        /// 45 Magenta
+        /// 46 Cyan
+        /// 47 White
+        /// </summary>
+        public void Parse(string message)
+        {
+            int startIndex = -1;
+            int length = 0;
+            ConsoleColor? foreground = null;
+            ConsoleColor? background = null;
+            var span = message.AsSpan();
+            const char EscapeChar = '\x1B';
+            ConsoleColor? color = null;
+            bool isBright = false;
+            for (int i = 0; i < span.Length; i++)
+            {
+                if (span[i] == EscapeChar && span.Length >= i + 4 && span[i + 1] == '[')
+                {
+                    if (span[i + 3] == 'm')
+                    {
+    #if (NETSTANDARD2_0 || NETFRAMEWORK)
```

Comment:
I thought we were going to flip this to be:

```
#if NETCOREAPP
  // net core stuff
#else
  // non-net core stuff
#endif
```

I think that approach is cleaner. Especially since we are not building for NETFRAMEWORK right now, it is confusing to have that define here.

Reply:
The reason why the `NETFRAMEWORK` was added here is because @joperezr is adding `net472` configurations in here: https://github.com/dotnet/runtime/pull/39099